providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/SemVerTest.java

```diff
@@ -1,186 +1,68 @@
 package dev.openfeature.contrib.providers.flagd.resolver.process.targeting;
 
 import io.github.jamsesso.jsonlogic.evaluator.JsonLogicEvaluationException;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
 
-import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class SemVerTest {
 
-    @Test
-    public void eqOperator() throws JsonLogicEvaluationException {
-        // given
-        final SemVer semVer = new SemVer();
-
-        // when
-        Object result = semVer.evaluate(Arrays.asList("v1.2.3", "=", "1.2.3"), new Object());
-
-        // then
-        if (!(result instanceof Boolean)) {
-            fail("Result is not of type Boolean");
-        }
-
-        assertThat((Boolean) result).isTrue();
-    }
-
-    @Test
-    public void neqOperator() throws JsonLogicEvaluationException {
-        // given
-        final SemVer semVer = new SemVer();
-
-        // when
-        Object result = semVer.evaluate(Arrays.asList("v1.2.3", "!=", "1.2.4"), new Object());
-
-        // then
-        if (!(result instanceof Boolean)) {
-            fail("Result is not of type Boolean");
-        }
-
-        assertThat((Boolean) result).isTrue();
+    static Stream<Arguments> validInputs() {
+        return Stream.of(
+                Arguments.of(Arrays.asList("v1.2.3", "=", "1.2.3")),
+                Arguments.of(Arrays.asList("v1.2.3", "!=", "1.2.4")),
+                Arguments.of(Arrays.asList("v1.2.3", "<", "1.2.4")),
+                Arguments.of(Arrays.asList("v1.2.3", "<=", "1.2.3")),
+                Arguments.of(Arrays.asList("v1.2.3", ">=", "v1.2.3")),
+                Arguments.of(Arrays.asList("v1.2.3", "^", "v1.0.0")),
+                Arguments.of(Arrays.asList("v5.0.3", "~", "v5.0.8")),
+                Arguments.of(Arrays.asList("v5.0.3", "~", "v5.0.8"))
+        );
     }
 
-    @Test
-    public void ltOperator() throws JsonLogicEvaluationException {
+    @ParameterizedTest
+    @MethodSource("validInputs")
```

Comment:
Did sonar DETECT this could be converted to a parameterized test??

Reply:
Yes it did. Sonar is smart ;)

Reply:
I'm impressed.